File: app/src/main/java/com/song/deviceinfo/info/HookInfo.java
package com.song.deviceinfo.info;

import android.content.Context;

import com.song.deviceinfo.utils.HookUtils;
import com.song.deviceinfo.utils.XposedHookUtils;

import java.util.ArrayList;
import java.util.List;

import androidx.core.util.Pair;

/**
 * Created by chensongsong on 2020/7/16.
 */
public class HookInfo {

    public static List<Pair<String, String>> getHookInfo(Context context) {
        List<Pair<String, String>> list = new ArrayList<>();
        list.add(new Pair<>("Process", HookUtils.checkRunningProcesses(context) + ""));
        list.add(new Pair<>("XposedHookMethod", HookUtils.chargeXposedHookMethod()));
        list.add(new Pair<>("XposedJars", HookUtils.chargeXposedJars()));
        list.add(new Pair<>("XposedPackage", HookUtils.chargeXposedPackage(context)));
        list.add(new Pair<>("XposedInject", XposedHookUtils.checkXposedInjet(context).toString()));
        list.add(new Pair<>("XposedClass", HookUtils.classCheck() + ""));
        return list;
    }

}

File: app/src/main/java/com/song/deviceinfo/info/HookInfo.kt
package com.song.deviceinfo.info

import android.content.Context
import androidx.core.util.Pair
import com.song.deviceinfo.utils.HookUtils.chargeXposedHookMethod
import com.song.deviceinfo.utils.HookUtils.chargeXposedJars
import com.song.deviceinfo.utils.HookUtils.chargeXposedPackage
import com.song.deviceinfo.utils.HookUtils.checkRunningProcesses
import com.song.deviceinfo.utils.HookUtils.classCheck
import com.song.deviceinfo.utils.XposedHookUtils.checkXposedInjet

/**
 * Created by chensongsong on 2020/7/16.
 */
object HookInfo {
    fun getHookInfo(context: Context): List<Pair<String, String>> {
        val list: MutableList<Pair<String, String>> = ArrayList()
        list.add(Pair("Process", checkRunningProcesses(context).toString() + ""))
        list.add(Pair("XposedHookMethod", chargeXposedHookMethod()))
        list.add(Pair("XposedJars", chargeXposedJars()))
        list.add(Pair("XposedPackage", chargeXposedPackage(context)))
        list.add(Pair("XposedInject", checkXposedInjet(context).toString()))
        list.add(Pair("XposedClass", classCheck().toString() + ""))
        return list
    }
}
